src/Hyperion/SerializerFactories/ExceptionSerializerFactory.cs

```diff
@@ -37,38 +37,53 @@ public ExceptionSerializerFactory()
 
         public override bool CanDeserialize(Serializer serializer, Type type) => CanSerialize(serializer, type);
 
+#if NETSTANDARD16
         // Workaround for CoreCLR where FormatterServices.GetUninitializedObject is not public
         private static readonly Func<Type, object> GetUninitializedObject =
             (Func<Type, object>)
                 typeof(string).GetTypeInfo().Assembly.GetType("System.Runtime.Serialization.FormatterServices")
                     .GetMethod("GetUninitializedObject", BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static)
                     .CreateDelegate(typeof(Func<Type, object>));
+#else
+        private object GetUninitializedObject(Type type) => System.Runtime.Serialization.FormatterServices.GetUninitializedObject(type);
+#endif
 
         public override ValueSerializer BuildSerializer(Serializer serializer, Type type,
             ConcurrentDictionary<Type, ValueSerializer> typeMapping)
         {
             var exceptionSerializer = new ObjectSerializer(type);
             var hasDefaultConstructor = type.GetTypeInfo().GetConstructor(new Type[0]) != null;
-            var createInstance = hasDefaultConstructor ? Activator.CreateInstance : GetUninitializedObject;
+            var exceptionObject = hasDefaultConstructor ? Activator.CreateInstance(type) : GetUninitializedObject(type);
 
             exceptionSerializer.Initialize((stream, session) =>
             {
-                var exception = createInstance(type);
+                var exception = exceptionObject;
```

Comment:
These two are not equivalent. `createInstance(type)` will return a new object on every deserializer call, while `exceptionObject` is an exception itself, so that the same instance will be captured and passed over and over. While this may not be obvious from simple unit tests, I'm not sure if this won't cause any troubles when called many times and from mutliple threads.

Do we have any way to prove that this change won't fail the running system?

Reply:
To be honest I did not test that scenario but you have a point. Therefore, I reverted it back to Func. Please see my updated pull request.